ert_shared/ensemble_evaluator/entity/ensemble_base.py

```diff
@@ -12,19 +14,76 @@
     SnapshotDict,
     Step,
 )
-from ert_shared.status.entity.state import (
-    ENSEMBLE_STATE_STARTED,
-    JOB_STATE_START,
-    REALIZATION_STATE_WAITING,
-    STEP_STATE_UNKNOWN,
-)
+import ert_shared.status.entity.state as state
+
+logger = logging.getLogger(__name__)
+
+
+class _EnsembleStateTracker:
```

Comment:
This looks more like a run state than an ensemble state - an ensemble is a meaningful object without concept of simulation and running.

Reply:
Yeah, agree—it should really be response or realisation.

Reply:
… but I don't think you should change this Dan. Even though the naming is a bit off, it's what we currently got. I can create an issue that could track implementation of proper naming, reflection on concepts, etc.